hnix-store-core/src/System/Nix/Store.hs

```diff
@@ -110,6 +66,7 @@ data StoreEffects rootedPath validPath m =
       derivationOutputNames :: !(validPath -> m (HashSet Text))
     , -- | Get a full 'Path' corresponding to a given 'Digest'.
       pathFromHashPart :: !(Digest PathHashAlgo -> m Path)
+    , narEffects :: NarEffects m
```

Comment:
We can move this around later, but I'm not sure it necessarily needs to be here. The `addNarToStore` effect can just take a `Nar`. We'll see.